frontend/src/scenes/billing/BillingLimitInput.tsx

```diff
@@ -20,16 +20,27 @@ export const BillingLimitInput = ({ product }: { product: BillingProductV2Type }
     )
     const { setIsEditingBillingLimit, setBillingLimitInput } = useActions(billingProductLogic({ product }))
 
+    const [billinglimitError, setBillingLimitError] = useState('')
+
     const updateBillingLimit = (value: number | undefined): any => {
         const actuallyUpdateLimit = (): void => {
             updateBillingLimits({
                 [product.type]: typeof value === 'number' ? `${value}` : null,
             })
         }
+
         if (value === undefined) {
-            return actuallyUpdateLimit()
+            // unset the limit
+            actuallyUpdateLimit()
+            return
+        }
+        if (isNaN(value) || value - Math.floor(value) !== 0) {
+            // Nan or decimal are not acceptable values.
+            setBillingLimitError('Please enter an integer.')
```

Comment:
I wonder how many non-engineers will wonder what an `integer` is... it's a bit of an easter egg for the engineering type so maybe we should keep it, or maybe we should just say "please enter a whole number"